### PR TITLE
Add project file Calculator.yashsachdev

### PR DESCRIPTION
**Pull request by yashsachdev**

### CODE REVIEW

**Comment by yashsachdev** [comment]
Project Title
Calcualtor.yashsachdev

Challenges
 1.Create a functionality that will count the amount of times the calculator was used.
 2.Store a list with the latest calculations. And give the users the ability to delete that list.
 3.Allow the users to use the results in the list above to perform new calculations.
 4.Add extra calculations: Square Root, Taking the Power, 10x, Trigonometry functions.

Intuition:

challenge 1 : ---------- 
challenge 2: Create a function which stores the list using .add() function and use .clear() function to delete.
	     Used the switch statement to let user access the function.
challenge 3:----------------

challenge 4: extra extra operators to switch, used System.math.


Approach:

challenge 1 :Created a field called _count increament count each time the function runs.
challenge 2: Hindsight switch statement might have been a wrong approach should have called those functions in a different way.
		Validate the index of the list to prevent Indexout of bound exception.
challenge 3:	Display the list and index using DisplayResult() and ask the user to enter the index. 
		and used the list[index] as input .

challenge 4: -------------------------------


@TheCSharpAcademy Thank you for your time and effort in reviewing my code and helping me learn. Any Feedback would be appreciated.

File: Calculator.yashsachdev/Calculator.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
+﻿namespace Calculator.yashsachdev;
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:1]: Please add one empty-line between a namespace declaration and a class.

File: Calculator.yashsachdev/Calculator.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+class Calculator
+{
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:4]: remove unnecessary empty line

File: Calculator.yashsachdev/Calculator.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            case "s":
+                result = num1 - num2;
+                _storeValue.Add(result);
+                break;
+            case "m":
+                result = num1 * num2;
+                _storeValue.Add(result);
+                break;
+            case "d":
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:37]: unnecessary empty line

File: Calculator.yashsachdev/Calculator.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                if (num2 != 0)
+                {
+                    result = num1 / num2;
+                    _storeValue.Add(result);
+                }
+                break;
+            //challenge 4:Add extra calculations: Square Root, Taking the Power, 10x, Trigonometry functions.
+            case "^":
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:45]: This one is a bit personal but I like to keep my switch statements lean. You could refactor this way for example. You can quickly refactor it by selecting a block of code and using "ctrl + ." or right-clicking and choosing quick refactorings. 
image

image

File: Calculator.yashsachdev/Calculator.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                _storeValue.Add(result);
+                break;
+            case "sin":
+                Console.WriteLine("Enter number you want to use for sin:");
+                double sinNum = Convert.ToDouble(Console.ReadLine());
+                result = Math.Sin(sinNum);
+                _storeValue.Add(result);
+                break;
+            case "1":
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:72]: unnecessary empty line

File: Calculator.yashsachdev/Calculator.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                break;
+
+            default:
+                break;
+        }
+        _count++;
+        Console.WriteLine($"The number of times Calculator was used {_count}"); //Challenge 1: Create a functionality that will count the amount of times the calculator was used.
+
+        return result;
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:89]: remove unnecessary 89,90 empty lines

File: Calculator.yashsachdev/Calculator.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    public void DisplayResult()
+    {
+        if (!_storeValue.Any())
+        {
+            Console.WriteLine("List is empty");
+        }
+
+        for (int i = 0; i < _storeValue.Count; i++)
+            Console.WriteLine("Result index=" + (i) + " :" + _storeValue[i]);
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Calculator.cs:115]: remove unnecessary empty line

File: Calculator.yashsachdev/Program.cs [hunk 8]
```diff
@@ -0,0 +1,146 @@
+﻿// See https://aka.ms/new-console-template for more information
+global using System;
+using Calculator;
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Program.cs:3]: Remove unused using.  
image

File: Calculator.yashsachdev/Program.cs [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+    else
+    {
+        Console.Write("Type another number OR type 'list' to use results from previous calculations , and then press Enter: ");
+    }
+
+    numInput2 = Console.ReadLine();
+    if (numInput2 == "list" && calculator.store.Count != 0) // Challenge 3: Allow the users to use the results in the list above to perform new calculations. 
+    {
+        calculator.DisplayResult(); //function to display list and its indexes.
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Program.cs:41]: It's ok to comment on code for learning, or when in development, but in general refrain from commenting, unless there's something here not explained by the code itself. If classes members (fields, methods, properties) are well-named, almost no comments are necessary.

File: Calculator.yashsachdev/Program.cs [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        Console.WriteLine("------------------------\n");
+        Console.Write("Press 'n' and Enter to close the app, or press any other key and Enter to continue: ");
+        if (Console.ReadLine() == "n") endApp = true;
+
+        Console.WriteLine("\n");
+    }
+
+    else
+    {
+
```
Comment (TheCSharpAcademy) [inline Calculator.yashsachdev/Program.cs:95]: unnecessary empty line

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]
It's looking great! I'm picking on the spacing issues because I like to cut these bad habits from the beginning. In terms of the code itself, the changes that are really important are the ones regarding refactoring your switch statements and commenting less.

**Pull request merged**